tf2_geometry_msgs/src/tf2_geometry_msgs/tf2_geometry_msgs.py

```diff
@@ -155,15 +154,146 @@ def transform_covariance(cov_in, transform):
     return cov_out.pose.covariance
 
 
+def _build_affine(
+        rotation: Optional[npt.ArrayLike] = None,
+        translation: Optional[npt.ArrayLike] = None) -> np.ndarray:
+    """
+    Build an affine matrix from a quaternion and a translation.
+
+    :param rotation: The quaternion as [w, x, y, z]
+    :param translation: The translation as [x, y, z]
+    :returns: The quaternion and the translation array
+    """
+    affine = np.eye(4)
+    if rotation is not None:
+        affine[:3, :3] = _get_mat_from_quat(np.asarray(rotation))
+    if translation is not None:
+        affine[:3, 3] = np.asarray(translation)
+    return affine
+
+
+def _transform_to_affine(transform: TransformStamped) -> np.ndarray:
+    """
+    Convert a `TransformStamped` to a affine matrix.
+
+    :param transform: The transform that should be converted
+    :returns: The affine transform
+    """
+    transform = transform.transform
+    transform_rotation_matrix = [
+        transform.rotation.w,
+        transform.rotation.x,
+        transform.rotation.y,
+        transform.rotation.z
+    ]
+    transform_translation = [
+        transform.translation.x,
+        transform.translation.y,
+        transform.translation.z
+    ]
+    return _build_affine(transform_rotation_matrix, transform_translation)
+
+
+def _get_mat_from_quat(quaternion: np.ndarray) -> np.ndarray:
+    """
+    Convert a quaternion to a rotation matrix.
+
+    This method is currently needed because transforms3d is not released as a `.dep` and
+    would require user interaction to set up.
+
+    For reference see: https://github.com/matthew-brett/transforms3d/blob/
+    f185e866ecccb66c545559bc9f2e19cb5025e0ab/transforms3d/quaternions.py#L101
```

Comment:
I think it might be less confusing to just say:

```
This method is a copy of quat2mat from https://github.com/matthew-brett/transforms3d/blob/f185e866ecccb66c545559bc9f2e19cb5025e0ab/transforms3d/quaternions.py#L101 , since that library is not available via rosdep.
```

Reply:
It is not an exact copy of the method (I changed some data types iirc), but I changed the description to `This method is based on ...` for `quad2mat` and used `This method is a copy of ...` for `mat2quad`, which is indeed the same code.